Features/header

lakhmanmaxim opened the pull request:
Я не могу никак дать ладу картинке "Рожок мороженного" - она изначально какая-то неправильная.
Жду замечаний.

lakhmanmaxim committed:
начало хедера

lakhmanmaxim committed:
Изменяет язык добавляет section

lakhmanmaxim committed:
мобльная и таблет версии готовы

мобльная и таблет версии готовы

lakhmanmaxim committed:
Merge branch 'main' into features/header

lakhmanmaxim committed:
Готово с примечанием

Примечание:
картинка мороженного - я не знаю как с ней быть.

KateDietkova commented on src/partials/hero-header.html:
```diff
@@ -0,0 +1,185 @@
+<Header class="header" id="home">
```
Виправ, будь ласка, тег header, з маленької літери напиши)

lakhmanmaxim replied:
Готово

KateDietkova commented on src/partials/hero-header.html:
```diff
@@ -0,0 +1,185 @@
+            </nav>
+            <!-- ====== Button Buy now ====== -->
+            <button class="mobile-menu-button--buy-now" type="button">Buy now
+                <svg class="mobile-menu-button__icon" width="10" height="11">
```
Розміри маленької свг стрілки 12*12, зміни, будь ласка) І клас свг зміни на  button__icon-arrow )

lakhmanmaxim replied:
Готово

KateDietkova commented on src/partials/hero-header.html:
```diff
@@ -0,0 +1,185 @@
+            <!-- ====== Button close ====== -->
+            <button type="button" class="mobile-menu-button--close" data-menu-close>
+                <svg width="32" height="32">
+                    <use href="./images/mobile-menu/burger.svg#mobile-menu-button--close"></use>
```
Скинь, будь ласка, у Slack burger.svg файл я ці свг внесу в загальний спрайт) Потім треба буде змінити шлях)

lakhmanmaxim replied:
Скинул

KateDietkova commented on src/partials/hero-header.html:
```diff
@@ -0,0 +1,185 @@
+    <div class="container container--header">
+        <a class="logo" href="./index.html">
+            <picture>
+                <source
```
Зміни, будь ласка, послідовність тегів source і виправ трошки медіазапити, спочатку пишеш source з media="(min-width: 1200px), потім з  media="(min-width: 768px), а потім media="(max-width: 767px) (цей медіазапит так і залишай з max-width: 767px) )

lakhmanmaxim replied:
Готово

KateDietkova commented on src/partials/hero-header.html:
```diff
@@ -0,0 +1,185 @@
+        </a>
+        <!-- ====== Button Mobile menu open ====== -->
+        <button type="button" class="mobile-menu-button--open" data-menu-open>
+            <svg width="20" height="12">
```
У цієї менюшки розміри повинні бути 32*32)

lakhmanmaxim replied:
Готово

KateDietkova commented on src/partials/hero-header.html:
```diff
@@ -0,0 +1,185 @@
+            </li>
+        </ul>
+        <!-- ====== Button buy now ====== -->
+        <button class="button--buy-now" type="button">Buy now
```
Задай, будь ласка, кнопці клас button (він вже прописаний у загальних стилях для неї) Модифікатор можеш залишити)

lakhmanmaxim replied:
Готово

KateDietkova commented on src/partials/hero-header.html:
```diff
@@ -0,0 +1,185 @@
+        </ul>
+        <!-- ====== Button buy now ====== -->
+        <button class="button--buy-now" type="button">Buy now
+            <svg class="mobile-menu-button__icon" width="10" height="11">
```
Розмір свг 12*12) І зміни тоді, будь ласка, ще назву класу свг на button__icon-arrow напиклад) Бо ця свг це елемент кнопки)

lakhmanmaxim replied:
готово

KateDietkova commented on src/partials/hero-header.html:
```diff
@@ -0,0 +1,185 @@
+            </svg>
+        </button>
+
+</Header>
```
header з маленької літери)

lakhmanmaxim replied:
готово

KateDietkova commented on src/partials/hero-header.html:
```diff
@@ -0,0 +1,185 @@
+
+</Header>
+
+<main>
```
main прибери, будь ласка, він прописується у index.html (я його пропишу) )

lakhmanmaxim replied:
готово

KateDietkova commented on src/partials/hero-header.html:
```diff
@@ -0,0 +1,185 @@
+                        ice cream made with <span class="hero-title hero__title--uppercase">passion</span>
+                    </h1>
+                    <ul class="hero__list">
+                        <li class="hero__item">
```
ul вважається як окремий блоковий елемент, тому його називаєш hero-list) Тобто незалежно від назви класу батьківського блоку)

lakhmanmaxim replied:
готово

KateDietkova commented on src/partials/hero-header.html:
```diff
@@ -0,0 +1,185 @@
+                        ice cream made with <span class="hero-title hero__title--uppercase">passion</span>
+                    </h1>
+                    <ul class="hero__list">
+                        <li class="hero__item">
```
li можеш задати тоді клас відповідно hero-list__item

lakhmanmaxim replied:
Done

KateDietkova commented on src/partials/hero-header.html:
```diff
@@ -0,0 +1,185 @@
+                    </h1>
+                    <ul class="hero__list">
+                        <li class="hero__item">
+                            <a class="hero__link hero__link--products" href="#products">Products</a>
```
У посилання будуть тоді класи hero-list__link та  hero-list__link--products )

lakhmanmaxim replied:
Done

KateDietkova commented on src/partials/hero-header.html:
```diff
@@ -0,0 +1,185 @@
+                </div>
+                <div class="hero-text-wrapper--addition">
+                    <ul class="hero-text-wrapper__list">
+                        <li class="hero-text-wrapper__item--bold">
```
У цьому списку йдуть тільки 2 лішки, і в кожній лішці по 2 p, виправ, будь ласка)

lakhmanmaxim replied:
Done

KateDietkova commented on src/partials/hero-header.html:
```diff
@@ -0,0 +1,185 @@
+                </div>
+                <!-- ====== DOTS ====== -->
+                <picture>
+                    <source
```
Так само, як і у попередньому тега picture я писала, зміни, будь ласка, послідовність тегів source і зміни медіазапити )

lakhmanmaxim replied:
Done

KateDietkova commented on src/partials/hero-header.html:
```diff
@@ -0,0 +1,185 @@
+                </picture>
+                <!-- ====== ICECREAM ====== -->
+                <picture>
+                    <source
```
Треба змінити послідовність source та змінити медіазапити )

lakhmanmaxim replied:
Done

KateDietkova commented on src/partials/hero-header.html:
```diff
@@ -0,0 +1,185 @@
+                    <img class="hero__image hero__image--icecream"
+                        src="./images/hero/ice-cream-mobile/ice-cream-mobile.png" alt="Ice cream">
+                </picture>
+                <div>
```
Тут пустий дів)

lakhmanmaxim replied:
Done

KateDietkova commented on src/partials/hero-header.html:
```diff
@@ -0,0 +1,185 @@
+                    et
+                </p>
+                <a href="#products">
+                    <svg width="40" height="15">
```
Тут достатньо задати width="40", height не потрібно)

lakhmanmaxim replied:
Решим, без 15 не выходило

lakhmanmaxim replied:
Пока оставил

KateDietkova commented on src/partials/hero-header.html:
```diff
@@ -0,0 +1,185 @@
+                </p>
+                <a href="#products">
+                    <svg width="40" height="15">
+                        <use href="./images/mobile-menu/burger.svg#big-arrow"></use>
```
Зміни будь ласка, шлях до загального свг спрайту я там додала white-arrow (але спочатку підтягни, будь ласка, зміни) )

lakhmanmaxim replied:
Вытяну новый спрайт и изменю пути.

KateDietkova commented on src/partials/hero-header.html:
```diff
@@ -0,0 +1,185 @@
+
+        </div>
+    </section>
+</main>
```
Прибери, будь ласка, main)

lakhmanmaxim replied:
Готово

KateDietkova commented on src/partials/hero-header.html:
```diff
@@ -0,0 +1,185 @@
+            <!-- ====== ICECREAM ====== -->
+            <picture>
+                <source
+                    srcset="./images/hero/plate-ice-cream-tablet/plate-ice-cream-tablet.png 1x, ./images/hero/plate-ice-cream-tablet/plate-ice-cream-tablet@2x.png 2x"
```
Так саамо, зміни, будь ласка, послідовність source, і виправ max-width: 1199px на min-width: 768px )

lakhmanmaxim replied:
Готово

KateDietkova commented on src/sass/layouts/_header.scss:
```diff
@@ -0,0 +1,149 @@
+  background-color: var(--primary-background-color);
+}
+
+.mobile-menu-button {
```
Зміни, будь ласка)
width: 32px;
 height: 32px;

lakhmanmaxim replied:
Готово

KateDietkova commented on src/sass/layouts/_header.scss:
```diff
@@ -0,0 +1,149 @@
+
+  margin-top: 20px;
+  padding: 0;
+  line-height: 0;
```
line-height: 0; зайве

lakhmanmaxim replied:
Готово

KateDietkova commented on src/partials/hero-header.html:
```diff
@@ -0,0 +1,185 @@
+                </ul>
+            </nav>
+            <!-- ====== Button Buy now ====== -->
+            <button class="mobile-menu-button--buy-now" type="button">Buy now
```
Тут задай додатково ще клас button модифікатор можеш поки залишити)

lakhmanmaxim replied:
Готово

KateDietkova commented on src/sass/layouts/_header.scss:
```diff
@@ -0,0 +1,149 @@
+  background: transparent;
+  cursor: pointer;
+
+  margin-top: 20px;
```
Від батьківського елемента не марджинами відштовхуємось, а падінгами самого батьківського елемента)

lakhmanmaxim replied:
Done

KateDietkova commented on src/sass/layouts/_header.scss:
```diff
@@ -0,0 +1,149 @@
+  background-color: var(--primary-background-color);
+}
+
+.mobile-menu-button {
```
Зверни увагу, що цей клас у розмітці в тебе ніде не задається) Це ок, просто треба у стилях тих модифікаторів кнопок що ти прописав у розмітці додати директиву @extend .mobile-menu-button

lakhmanmaxim replied:
Done

KateDietkova commented on src/sass/layouts/_header.scss:
```diff
@@ -0,0 +1,149 @@
+
+.logo {
+  display: block;
+  margin-top: 12px;
```
Замість марджина задай верхній падінг хедеру в якому знаходиться лого, навігація і кнопка)

lakhmanmaxim replied:
Done

KateDietkova commented on src/sass/layouts/_header.scss:
```diff
@@ -0,0 +1,149 @@
+  margin-top: 12px;
+}
+
+.container {
```
Прибери будь ласка, це правило, стилі контейнера вже є у загальних)

lakhmanmaxim replied:
Done

KateDietkova commented on src/sass/layouts/_header.scss:
```diff
@@ -0,0 +1,149 @@
+  background-color: var(--primary-background-color);
+}
+
+.container-hero {
```
В тебе це клас не приміняється, бо в тебе у розмітці прописаний саме модифікатор container--hero, можеш просто змінити тут назву)

lakhmanmaxim replied:
Done

KateDietkova commented on src/partials/hero-header.html:
```diff
@@ -0,0 +1,185 @@
+                        </li>
+                    </ul>
+                </div>
+                <div class="hero-text-wrapper--addition">
```
Тут дів зайвий, достатньо і  ul, він також є блоковим елементом)

lakhmanmaxim replied:
Done

KateDietkova commented on src/sass/layouts/_header.scss:
```diff
@@ -0,0 +1,149 @@
+  overflow: hidden;
+}
+
+.hero-title-container {
```
Цей клас ніде не використовується) Видали його, будь ласка)

lakhmanmaxim replied:
DOne

KateDietkova commented on src/sass/layouts/_header.scss:
```diff
@@ -0,0 +1,149 @@
+
+.hero-title {
+  font-family: var(--font-title);
+  font-weight: 400;
```
font-weight: 400; це можна не зазначати, оскільки по дефолту повинно ставитись)

lakhmanmaxim replied:
Почистил

KateDietkova commented on src/sass/layouts/_header.scss:
```diff
@@ -0,0 +1,149 @@
+  letter-spacing: 0.04em;
+  text-transform: uppercase;
+
+  color: var(--secondary-lighttext-color);
```
Така змінна має сірий колір, виправ на --primary-lighttext-color)

lakhmanmaxim replied:
Готово

KateDietkova commented on src/sass/layouts/_header.scss:
```diff
@@ -0,0 +1,149 @@
+  line-height: 0.97;
+}
+
+.hero-list {
```
Оскільки це список треба прописати сюди наслідування від плейсхолдера @extend %list; ) І видали margin: 0;
  padding: 0;
  list-style: none; це все скидається вже у базових налаштуваннях) І зверни увагу, що такого класу в тебе поки, що нема у розмітці, його треба задати замість hero__list, я писала зверху)

lakhmanmaxim replied:
Done

KateDietkova commented on src/sass/layouts/_header.scss:
```diff
@@ -0,0 +1,149 @@
+}
+
+.hero-link {
+  font-family: var(--font-general);
```
Оскільки це посилання треба зробити наслідування від плейсхолдера extend %link; та прибрати text-decoration: none;

lakhmanmaxim replied:
DOne

KateDietkova commented on src/sass/layouts/_header.scss:
```diff
@@ -0,0 +1,149 @@
+  text-decoration: none;
+}
+
+.hero-link--products {
```
В тебе нема такого класу у розмітці, видали його)

lakhmanmaxim replied:
DOne

KateDietkova commented on src/sass/layouts/_hero-header.scss:
```diff
@@ -0,0 +1,48 @@
+    display: none;
+  }
+
+  @media screen and (min-width: 768px) {
```
Цей медіазапит зі стилями можеш прибрати) Це все описано у загальних стилях button) А margin top не кнопці треба ставити, треба хедеру задати підінг до найбільшого у ньому елемента)

lakhmanmaxim replied:
Done

KateDietkova commented on src/sass/layouts/_hero-header.scss:
```diff
@@ -0,0 +1,48 @@
+.container--header {
+  position: relative;
+  display: flex;
+  justify-content: end;
```
Сюди спробуй задати  align-items: center;

lakhmanmaxim replied:
Done

KateDietkova commented on src/sass/layouts/_hero-header.scss:
```diff
@@ -0,0 +1,48 @@
+    font-size: 14px;
+    line-height: 1.21;
+
+    margin-top: 24px;
```
margin top не треба ставити, треба хедеру задати підінг до найбільшого у ньому елемента)

KateDietkova commented on src/sass/layouts/_hero-header.scss:
```diff
@@ -0,0 +1,48 @@
+    line-height: 1.21;
+
+    margin-top: 24px;
+    margin-left: 31px;
```
margin-left: 32px; треба поставити і заключити його в медіазапит max-width: 1199px

lakhmanmaxim replied:
Done

KateDietkova commented on src/sass/layouts/_hero-header.scss:
```diff
@@ -0,0 +1,48 @@
+  @media screen and (min-width: 1200px) {
+    padding: 12px 30px;
+
+    margin-top: 28px;
```
Так само тут всі мерджини прибери)

KateDietkova commented on src/sass/layouts/_hero-header.scss:
```diff
@@ -0,0 +1,48 @@
+    margin-left: 31px;
+  }
+
+  @media screen and (min-width: 1200px) {
```
Це медіаправило також потрібно прибрати)

lakhmanmaxim replied:
Done

KateDietkova commented on src/partials/hero-header.html:
```diff
@@ -0,0 +1,185 @@
+<Header class="header" id="home">
```
header з маленької)

lakhmanmaxim replied:
Done

KateDietkova reviewed (changes requested):
Вже виправили) Чекаю нового коміту))

lakhmanmaxim committed:
Корректировки по Pull Request

lakhmanmaxim committed:
Merge branch 'main' into features/header

lakhmanmaxim committed:
Изменил спрайт и ссылки на него

1. Изменил спрайт и ссылки на него в коде.
2. Немного поправил позиционирование кнопки закрытия мобильного меню.
3. Свой спрайт удалил.

lakhmanmaxim committed:
Header+hero = done

Header+hero = done
Только ссылка How it's made подпрыгнула - паддинга нет.

lakhmanmaxim committed:
Merge branch 'main' into features/header

lakhmanmaxim committed:
Маленькая правка после валидатора

Маленькая правка после валидатора - не закрытый ДИВ

lakhmanmaxim committed:
Merge branch 'main' into features/header

lakhmanmaxim committed:
Merge branch 'main' into features/header

lakhmanmaxim committed:
Корректировка

 Корректировка паддинга и привязки кнопки закрытия моб. меню

lakhmanmaxim committed:
Merge branch 'main' into features/header
